pyresample/test/test_utils.py

```diff
@@ -724,3 +726,51 @@ def test_check_slice_orientation():
     slice_in = slice(start, stop, step)
     res = check_slice_orientation(slice_in)
     assert res == slice(start, stop, -1)
+
+
+class TestRowAppendableArray(unittest.TestCase):
+    """Test appending numpy arrays to possible pre-allocated buffer."""
+
+    def test_append_1d_arrays_and_trim_remaining_buffer(self):
+        appendable = RowAppendableArray(7)
+        appendable.append_row(np.zeros(3))
+        appendable.append_row(np.ones(3))
+        self.assertTrue(np.array_equal(appendable.to_array(), np.array([0, 0, 0, 1, 1, 1])))
+
+    def test_append_rows_of_nd_arrays_and_trim_remaining_buffer(self):
+        appendable = RowAppendableArray(7)
+        appendable.append_row(np.zeros((3, 2)))
+        appendable.append_row(np.ones((3, 2)))
+        self.assertTrue(np.array_equal(appendable.to_array(), np.vstack([np.zeros((3, 2)), np.ones((3, 2))])))
+
+    def test_append_more_1d_arrays_than_expected(self):
+        appendable = RowAppendableArray(5)
+        appendable.append_row(np.zeros(3))
+        appendable.append_row(np.ones(3))
+        self.assertTrue(np.array_equal(appendable.to_array(), np.array([0, 0, 0, 1, 1, 1])))
+
+    def test_append_more_rows_of_nd_arrays_than_expected(self):
+        appendable = RowAppendableArray(2)
+        appendable.append_row(np.zeros((3, 2)))
+        appendable.append_row(np.ones((3, 2)))
+        self.assertTrue(np.array_equal(appendable.to_array(), np.vstack([np.zeros((3, 2)), np.ones((3, 2))])))
+
+    def test_append_1d_arrays_pre_allocated_appendable_array(self):
+        appendable = RowAppendableArray(6)
+        appendable.append_row(np.zeros(3))
+        appendable.append_row(np.ones(3))
+        self.assertTrue(np.array_equal(appendable.to_array(), np.array([0, 0, 0, 1, 1, 1])))
+
+    def test_append_rows_of_nd_arrays_to_pre_allocated_appendable_array(self):
+        appendable = RowAppendableArray(6)
+        appendable.append_row(np.zeros((3, 2)))
+        appendable.append_row(np.ones((3, 2)))
+        self.assertTrue(np.array_equal(appendable.to_array(), np.vstack([np.zeros((3, 2)), np.ones((3, 2))])))
+
+    def test_pre_allocation_can_double_appending_performance(self):
+        unallocated = RowAppendableArray(0)
+        pre_allocated = RowAppendableArray(10000)
+
+        unallocated_performance = timeit(lambda: unallocated.append_row(np.array([42])), number=10000)
+        pre_allocated_performance = timeit(lambda: pre_allocated.append_row(np.array([42])), number=10000)
+        self.assertGreater(unallocated_performance / pre_allocated_performance, 2)
```

Comment:
I'm not sure how I feel about a unit test that checks performance...but maybe that's OK. Any idea how long this test takes to run?

Reply:
On my very dated machine with an i5-6500 CPU @ 3.20GHz with 4 physical cores, it takes 88ms. 

I admit that this is probably a bit overzealous here, as we can safely assume that preallocation will improve performance, it just checks that preallocation is actually happening. Thinking about it now, it might be better to remove it since it runs the risk of making the test suite brittle.

Running on gitlab workers where you don't have that much control over load, hardware etc. when the test runs makes creating performance tests hard. I just felt that this one is quite on the safe side since it checks relative performance and we can assume that preallocation gives significant gains on most systems no matter the load, however, it might not be worth it.

Reply:
Good enough for me. Thanks. Let's get this merged and see how people feel about it. Thanks for all the work and back and forth with me.